compiler/aarch64/codegen/GenerateInstructions.cpp

```diff
@@ -250,3 +250,31 @@ TR::Instruction *generateLogicalShiftLeftImmInstruction(TR::CodeGenerator *cg, T
       return new (cg->trHeapMemory()) TR::ARM64Trg1Src1ImmInstruction(op, node, treg, sreg, imm, preced, cg);
    return new (cg->trHeapMemory()) TR::ARM64Trg1Src1ImmInstruction(op, node, treg, sreg, imm, cg);
    }
+
+TR::Instruction *generateCompareImmInstruction(TR::CodeGenerator *cg, TR::Node *node,
+   TR::Register *sreg, int32_t imm, TR::Instruction *preced)
+   {
+   /* Alias of SUBS instruction */
+
+   bool is64bit = node->getDataType().isInt64();
+   TR::InstOpCode::Mnemonic op = is64bit ? TR::InstOpCode::Mnemonic::subsimmx : TR::InstOpCode::Mnemonic::subsimmw;
+   TR::RealRegister *zeroReg = cg->machine()->getARM64RealRegister(TR::RealRegister::xzr);
+
+   if (preced)
+      return new (cg->trHeapMemory()) TR::ARM64Trg1Src1ImmInstruction(op, node, zeroReg, sreg, imm, preced, cg);
+   return new (cg->trHeapMemory()) TR::ARM64Trg1Src1ImmInstruction(op, node, zeroReg, sreg, imm, cg);
+   }
+
+TR::Instruction *generateTestImmInstruction(TR::CodeGenerator *cg, TR::Node *node,
+   TR::Register *sreg, int32_t imm, TR::Instruction *preced)
+   {
+   /* Alias of ANDS instruction */
+
+   bool is64bit = node->getDataType().isInt64();
+   TR::InstOpCode::Mnemonic op = is64bit ? TR::InstOpCode::Mnemonic::andsimmx : TR::InstOpCode::Mnemonic::andsimmw;
+   TR::RealRegister *zeroReg = cg->machine()->getARM64RealRegister(TR::RealRegister::xzr);
```

Comment:
Will the AArch64 register assigner be able to handle assigning real registers attached to the instruction before register assignment?  I'm not saying this is a problem, just confirming that's the case.

Reply:
I am not sure how it works.
Is there any suggested way to use a fixed register (`xzr` in this case) in `GenerateXXXInstruction()` ?

Reply:
Nothing seems to prevent calling `getARM64RealRegister(TR::RealRegister::xzr)`.  `xzr` is never used by the register assignment for aarch64.

Reply:
Yes.  Thanks Konno-san.

Reply:
I added some constructors with register dependencies for `xzr` register association.
I may need to add similar constructors for ARM64Trg1Src2Instruction and ARM64Trg1Src3 Instruction when needed.